cli/lib/kontena/cli/stacks/yaml/validations.rb

```diff
@@ -43,7 +43,7 @@ def common_validations
               )
             end
           elsif value.is_a?(Array)
-            value.all? { |v| v.kind_of?(String) && v =~ /\A\S+(?<!\\)=.*/ }
+            value.all? { |v| v.kind_of?(String) && v =~ /^([^=]+)=(.*)/ }
```

Comment:
OTOH there's no reason to not accept multi-line env values, so I suggest `/\A\S+=/`

```yaml
stack: test/test
services:
  foo:
    environment:
      - |
          FOO=
          ASDF1
          ASDF2
```

```
            "Env": [
                "FOO=\nASDF1\nASDF2",
            ],
```

```
$ echo $FOO
ASDF1 ASDF2
```